src/main/java/httpServer/ListenerAndResponder.java

```diff
@@ -40,46 +43,114 @@ public Socket keepListening() throws IOException {
     }
 
     public void readClientInput(Socket clientSocket) throws IOException {
-
+        System.err.println("Starting to read");
         ClientReadable bufferedReader = clientReaderFactory.makeReader(clientSocket);
-
-        String httpRequest;
-        httpRequest = readAllLines(bufferedReader);
+        HTTPRequest httpRequest;
+        httpRequest = readAllLinesAndReturnHttpRequest(bufferedReader);
         if (httpRequest == null) {
             StatusCodes statusCodes = StatusCodes.BAD_REQUEST;
             String responseText = statusCodes.httpResponse;
             printServerResponse(responseText, clientSocket);
         }
         if (!httpRequest.equals("")) {
-            HTTPRequest request = RequestParser.parse(httpRequest);
-            System.out.println(httpRequest);
-            String httpResponse = router.getResponse(request);
-            printServerResponse(httpResponse, clientSocket);
+            HTTPResponse httpResponse = router.getResponse(httpRequest);
+            String responseString = httpResponse.toString();
+            System.err.println(responseString.length());
```

Comment:
Do you still need this statement or is it for debugging?

Reply:
Debugging! Oops, fixed